### PR TITLE
Use modern java idiom

### PR DESCRIPTION
**Pull request by joe-chacko**
#build

### CODE REVIEW

**Comment by joe-chacko** [comment]
#build

**Comment by joe-chacko** [comment]
That build failed checking out related source, so let's #build again.

**Comment by joe-chacko** [comment]
Manually requested a build here: 
https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_PNcWYCzREe6bFfaWrnDnjQ

**Review by habiblawal1 — DISMISSED** [review]

**Comment by joe-chacko** [comment]
#build

**Comment by joe-chacko** [comment]
#build

File: dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java [hunk 1]
```diff
@@ -125,35 +130,29 @@ protected void deactivate() {
...
-        }
+        Optional.ofNullable((String) properties.get("orbSSLInitTimeout"))
+                .filter(s -> !s.isEmpty())
+                .map(Long::valueOf)
+                .ifPresent(timeout -> {
+                    if (isAnyTracingEnabled() && tc.isDebugEnabled())
+                        debug(tc, "TIMEOUT_SECONDS = " + TIMEOUT_SECONDS);
+                    TIMEOUT_SECONDS = timeout;
+                });
+
```
Comment (Azquelt) [inline dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java]: This is correct but I think I prefer the previous code.

File: dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java [hunk 2]
```diff
@@ -210,21 +204,15 @@ public ReadyRegistration(Set<String> requiredSslRefs, ReadyListener listener) {
...
 
         void check() {
             boolean containsAll = AbstractCsiv2SubsystemFactory.this.check(requiredSslRefs);
             listener.readyChanged(AbstractCsiv2SubsystemFactory.this, containsAll);
-            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                Tr.debug(tc, "Check: Known ssl configurations: {0}, required: {1}, containsAll: {2} timeout exists: {3}", sslRefs, requiredSslRefs, containsAll, future != null);
-            }
+            if (isAnyTracingEnabled() && tc.isDebugEnabled())
+                debug(tc, "Check: Known ssl configurations: {0}, required: {1}, containsAll: {2} timeout exists: {3}",
+                        sslRefs, requiredSslRefs, containsAll, future != null);
```
Comment (Azquelt) [inline dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java]: This is correct, but in general I would leave the braces in.

File: dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/config/ssl/yoko/SSLSubsystemFactory.java [hunk 3]
```diff
@@ -86,20 +79,14 @@ public Policy getTargetPolicy(ORB orb, Map<String, Object> properties, Map<Strin
...
     private boolean isAnySecureTransportAddressAvailable(Map<String, Object> extraConfig) {
-        Map<String, List<TransportAddress>> addrMap = (Map<String, List<TransportAddress>>) extraConfig.get(ADDR_KEY);
-        if (addrMap != null) {
-            Set<String> sslAliases = addrMap.keySet();
-            return sslAliases.size() != 1;
-        }
-        return true;
+        Object addrMap = extraConfig.get(ADDR_KEY);
+        if (null == addrMap) return true;
+        return ((Map<?,?>)addrMap).keySet().size() != 1;
```
Comment (Azquelt) [inline dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/config/ssl/yoko/SSLSubsystemFactory.java]: `Map` itself has a `size` method which I think would always return the same value as the size of the key set?

File: dev/com.ibm.ws.security.csiv2/src/com/ibm/ws/transport/iiop/server/security/CSIv2SubsystemFactory.java [hunk 4]
```diff
@@ -224,25 +215,22 @@ private Map<String, List<TransportAddress>> extractTransportAddresses(Map<String
...
-            }
-        }
+        // add all the sslRef aliases (and defaultAlias if they are null)
+        Set<String> result = endpoints.stream()
+                .map(IIOPEndpoint::getIiopsOptions)
+                .flatMap(List::stream)
+                .map(map -> (String) map.get("sslRef"))
+                .map(alias -> null == alias ? defaultAlias : alias)
+                .collect(HashSet::new, Set::add, Set::addAll);
+        //
```
Comment (Azquelt) [inline dev/com.ibm.ws.security.csiv2/src/com/ibm/ws/transport/iiop/server/security/CSIv2SubsystemFactory.java]: Last line could be
```java
.collect(toCollection(HashSet::new))
```
using `Collections.toCollection`.

Also there's a stray empty comment here.

**Review by Azquelt — CHANGES_REQUESTED** [review]
I think the copyright dates on these classes should be updated.

Otherwise it looks good though I've left a few comments for your consideration.

File: dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java (hunk 1 above)
Comment (joe-chacko) [reply dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java]: reverted

File: dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java (hunk 2 above)
Comment (joe-chacko) [reply dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/AbstractCsiv2SubsystemFactory.java]: Done

File: dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/config/ssl/yoko/SSLSubsystemFactory.java (hunk 3 above)
Comment (joe-chacko) [reply dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/config/ssl/yoko/SSLSubsystemFactory.java]: Of course — will use that instead.

Comment (joe-chacko) [reply dev/com.ibm.ws.security.csiv2.common/src/com/ibm/ws/transport/iiop/security/config/ssl/yoko/SSLSubsystemFactory.java]: Didn't reactor quite far enough…

File: dev/com.ibm.ws.security.csiv2/src/com/ibm/ws/transport/iiop/server/security/CSIv2SubsystemFactory.java (hunk 4 above)
Comment (joe-chacko) [reply dev/com.ibm.ws.security.csiv2/src/com/ibm/ws/transport/iiop/server/security/CSIv2SubsystemFactory.java]: That's handy! Done.

Comment (joe-chacko) [reply dev/com.ibm.ws.security.csiv2/src/com/ibm/ws/transport/iiop/server/security/CSIv2SubsystemFactory.java]: **`Collectors.`**`toCollection()` though.
That foxed me briefly.

**Comment by joe-chacko** [comment]
Made reviewer-requested changes.
#build and #run-libby-bot again

File: dev/com.ibm.ws.security.csiv2/src/com/ibm/ws/transport/iiop/server/security/CSIv2SubsystemFactory.java (hunk 4 above)
Comment (Azquelt) [reply dev/com.ibm.ws.security.csiv2/src/com/ibm/ws/transport/iiop/server/security/CSIv2SubsystemFactory.java]: Ack, sorry :(

**Review by Azquelt — APPROVED** [review]
Looks good :+1:

**Comment by joe-chacko** [comment]
#build again — the previous one had environmental issues

**Pull request merged**